py/util.h

```diff
@@ -44,14 +44,19 @@ convert_to_double( PyObject* obj, double& out )
 inline bool
 convert_to_strength( PyObject* value, double& out )
 {
+  std::string str;
+
 #if PY_MAJOR_VERSION >= 3
     if( PyUnicode_Check( value ) )
     {
-        std::string str( PyUnicode_AsUTF8( value ) );
+      str = PyUnicode_AsUTF8( value );
 #else
-    if( PyString_Check( value ) )
+    if( PyString_Check( value ) | PyUnicode_Check( value ))
     {
-        std::string str( PyString_AS_STRING( value ) );
+      if( PyUnicode_Check( value ) )
+          str = PyString_AS_STRING(PyUnicode_AsASCIIString( value ) );
```

Comment:
Why not use PyUnicode_AsUTF8 as we do on Python 3 ?

Reply:
Yeah. `PyUnicode_AsASCIIString` also returns a new object, so this line will cause a memory leak.
https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_AsASCIIString